### PR TITLE
GUACAMOLE-1256: Correct terminal scroll with text editors.

### PR DESCRIPTION
**Pull request by corentin-soriano**
- Add support of 'S' and 'T' CSI sequences (scroll up/down by amount).
- Force display flush when scrolling to avoid conflict when cursor_commit use the set_collumns function which change the operation_type.

### CODE REVIEW

File: src/terminal/terminal-handlers.c [hunk 1]
```diff
@@ -835,6 +835,28 @@ int guac_terminal_csi(guac_terminal* term, unsigned char c) {
...
                 break;
 
+            /* Scroll Up by amount */
+            case 'S':
+
+                /* Multiple linefeed */
+                for (int i = 0; i < argv[0]; i++)
+                    guac_terminal_linefeed(term, false);
+
+                term->char_handler = guac_terminal_echo;
```
Comment (necouchman) [inline src/terminal/terminal-handlers.c:845]: Why is this required? I don't see this assignment required in any of the other `case` statements - is there some reason why it needs to be assigned for these two new instances?

Comment (corentin-soriano) [reply src/terminal/terminal-handlers.c]: Yes you are right, it is not useful since it is already assigned at line 1117 (in this branch).
I removed them.

File: src/terminal/terminal-handlers.c [hunk 2]
```diff
@@ -835,6 +835,24 @@ int guac_terminal_csi(guac_terminal* term, unsigned char c) {
 
                 break;
 
+            /* Scroll Up by amount */
+            case 'S':
+
+                /* Multiple linefeed */
+                for (int i = 0; i < argv[0]; i++)
+                    guac_terminal_linefeed(term, false);
```
Comment (necouchman) [inline src/terminal/terminal-handlers.c:843]: One other question - above, in the sections for inserting/deleting lines, there are functions `guac_terminal_scroll_up` and `guac_terminal_scroll_down` - is there a reason not to use these, versus using `for` loops you have here?

Comment (corentin-soriano) [reply src/terminal/terminal-handlers.c]: Yes, we can use these functions directly instead of doing a for loop.
I changed and tested again: the result is the same.

**Comment by hugaleno** [comment]
Thank you once again @corentin-soriano. Another amazing contribution

**Review by necouchman — APPROVED** [review]
